protocol/contracts/v2/safeGuards/ThresholdSafeGuard.sol

```diff
@@ -0,0 +1,236 @@
+        TempleElevatedAccess(_initialRescuer, _initialExecutor)
+    {
+        defaultSignaturesThreshold = _defaultSignaturesThreshold;
+    }
+
+    /**
+     * @notice Disable the Safe Guard, so all transactions pass if the Gnosis threshold of signers is met.
+     *   1. Rescuers setRescueMode(true)
+     *   2. Rescuers setDisableGuardChecks(true)
+     * Note: Rescuers won't have a Guard on their multisig, so this can't be bricked.
+     */
+    function setDisableGuardChecks(bool value) external onlyInRescueMode {
+        disableGuardChecks = value;
+        emit DisableGuardChecksSet(value);
+    }
+
+    /**
+      * @notice Add an address to the allowed list who can execute the transaction
+      * once the minimum threshold of signers have approved
+     */
+    function addSafeTxExecutor(address executor) external onlyElevatedAccess {
+        if (executor == address(0)) revert InvalidAddress();
+        emit SafeTxExecutorAdded(executor);
+        if (!_safeTxExecutors.add(executor)) revert InvalidExecutor();
+    }
+
+    /**
+      * @notice Remove an address from the allowed list who can execute the transaction
+      * once the minimum threshold of signers have approved
+     */
+    function removeSafeTxExecutor(address executor) external onlyElevatedAccess {
+        if (executor == address(0)) revert InvalidAddress();
+        emit SafeTxExecutorRemoved(executor);
+        if (!_safeTxExecutors.remove(executor)) revert InvalidExecutor();
+    }
+
+    /**
+      * @notice Set the default number of signatories required, when a contract/function signature
+      * not explicitly defined.
+     */
+    function setDefaultSignaturesThreshold(uint256 threshold) external onlyElevatedAccess {
+        emit DefaultSignaturesThresholdSet(threshold);
+        defaultSignaturesThreshold = threshold;
+    }
+
+    /**
+      * @notice Set the number of signatories required for a contract/function signature pair.
+     */
+    function setFunctionThreshold(address contractAddr, bytes4 functionSignature, uint256 threshold) external onlyElevatedAccess {
+        if (contractAddr == address(0)) revert InvalidAddress();
+        if (functionSignature == bytes4(0)) revert InvalidFunctionSignature();
+
+        emit FunctionThresholdSet(contractAddr, functionSignature, threshold);
+        functionThresholds[contractAddr][functionSignature] = threshold;
+    }
+
+    /**
+      * @notice Set the number of signatories required for a number of function signatures at a time.
+     */
+    function setFunctionThresholdBatch(address contractAddr, bytes4[] memory functionSignatures, uint256 threshold) external onlyElevatedAccess {
+        if (contractAddr == address(0)) revert InvalidAddress();
+
+        uint256 length = functionSignatures.length;
+        bytes4 sig;
+        for (uint256 i; i < length; ++i) {
+            sig = functionSignatures[i];
+            if (sig == bytes4(0)) revert InvalidFunctionSignature();
```

Comment:
don't you rather want to skip instead of reverting?

Reply:
No i'd rather fail the whole thing (it would likely be a tenderly sim which runs first for advance notification) so they correct the issue